src/language/ast.js

```diff
@@ -398,15 +403,15 @@ export type ObjectTypeDefinitionNode = {
   name: NameNode,
   interfaces?: ?Array<NamedTypeNode>,
   directives?: ?Array<DirectiveNode>,
-  fields: Array<FieldDefinitionNode>,
+  fields?: ?Array<FieldDefinitionNode>,
```

Comment:
We aren't going to support the empty-set definition, are we? Because it's still defined in the SDL as
```
ObjectTypeDefinition : Description? type Name ImplementsInterfaces? Directives[Const]? FieldDefinitions?

FieldDefinitions : { FieldDefinition+ }
```

So this should probably just be
```
fields?: Array<FieldDefinitionNode>,
```
right? Or is there a reason allowing it to be both unset and nullable is an advantage I'm missing?

Reply:
Actually I think due to recent Flow changes we may be able to do this instead.

Previously `field?: Array<FieldDefinition>` meant that if `field` was defined, it *must* be an Array, but now I think it means `field` may be defined with the value `undefined` which is very useful for cases where you're writing a utility like `{field: otherThing.field}`. I'll investigate that and follow up with a separate PR

Reply:
@mjmahone I tightened up the AST definitions as you suggested plus made them covariant in #1121